docs/user/agent/authentication-authorization.md

```diff
@@ -0,0 +1,124 @@
+# Agent-Based Installer Authentication and Authorization - OpenShift 4.18
+
+## Introduction
+In OpenShift 4.18, the Assisted Service APIs used by the Agent-Based Installer are now secured through authentication and authorization mechanisms. This added security ensures that only authorized services can interact with the installer and other system components. This document provides an overview of how to enable authentication and authorization for the agent-based installer.
+
+## Enabling Authentication and Authorization
+
+To enable authentication and authorization, the `AuthType` setting must be configured in the `assisted-service.env.template` file. Set the following value:
+
+```bash
+AuthType = "agent-installer-local"
+```
+
+This setting ensures that the authentication mechanism required by the agent-based installer is activated.
+
+## Tokens for Agent-Based Installer
+
+When creating an Agent ISO or PXE files on Day 1, the OpenShift installer generates three separate authentication tokens: `agentAuth`, `userAuth`, and `watcherAuth`. These tokens serve different purposes and have varying levels of access.
+
+### Types of Tokens:
+1. **agentAuth**:
+   - **Access Level**: Read-Write (RW)
+   - **Used By**: The agent service that runs on hosts booted with the agent-based installer.
+   
+2. **userAuth**:
+   - **Access Level**: Read-Write (RW)
+   - **Used By**: Systemd services and the agent-based installer client in the Assisted Service.
+   
+3. **watcherAuth**:
+   - **Access Level**: Read-Only (RO)
+   - **Used By**: Only the `wait-for` command (on Day 1) and `monitor-add-nodes` command (on Day 2).
+
+### Token Expiry Behavior
+- **Day 1** Tokens: The tokens generated on Day 1 do not expire, meaning the agent ISO remains valid indefinitely. This is useful for use cases where customers create an agent ISO on one day and boot it on a later day (e.g., in a data center).
+- **Day 2** Tokens: Tokens generated for nodes ISO on Day 2 expire after 48 hours. This ensures that the nodes.iso is only valid for a short period.
+
+### Token Management
+When a Day 2 node ISO is created, all three tokens are saved in the `openshift-config` namespace as a new secret named `agent-auth-token`. The following steps outline how token management works:
+1. If the secret exists in the cluster, the token can be retrieved.
+2. All tokens expire at the same time, so only one token needs to be checked for expiry.
+3. The expiration time is calculated as 24 hours before the token's expiry time.
+4. If the token in the secret is older than 24 hours when creating a node ISO, the installer generates new tokens. These new tokens are saved in the asset store, and the secret in the cluster is updated with the new token values.
+5. If the token in the secret is not older than 24 hours, the installer retrieves the existing token from the cluster, updates the asset store with the retrieved token, and uses the retrieved token for the new node ISO.
```

Comment:
This looks like something that could be also shown better using a simple diagram. Have you maybe ever try [mermaid](https://github.com/mermaid-js/mermaid) for that?

Reply:
Sure, I will add it.